src/OVAL/probes/independent/system_info.c

```diff
@@ -343,11 +626,14 @@ void *probe_init(void)
 int probe_main(probe_ctx *ctx, void *arg)
 {
 	SEXP_t* item;
-	char* os_name, *os_version, *architecture, *hname;
+	char* os_name, *architecture, *hname;
+	const char *os_version = NULL;
 	struct utsname sname;
 	probe_offline_flags offline_mode = PROBE_OFFLINE_NONE;
 	(void)arg;
+	char unknown[] = "Unknown";
```

Comment:
Shouldn't this be `const char *unknown = "Unknown";`? I would also just `strdup` it to avoid the logic of freeing or not freeing the entries.